plugins/pipx/plugin.json

```
@@ -0,0 +1,15 @@
+{
+  "name": "pipx",
+  "version": "0.1.0",
+  "description": "Installs Pipx and makes it available",
+  "packages": ["pipx@latest"],
+  "env": {
+    "PIPX_HOME": "{{ .Virtenv }}",
+    "PIPX_BIN_DIR": "{{ .Virtenv }}/bin"
+  },
+  "shell": {
+      "init_hook": [
+          "export PATH=\"{{ .Virtenv }}/bin:$PATH\""
+      ]
+  }
```

[suggestion]: you can update the path without an `init_hook`
```suggestion
    "PIPX_BIN_DIR": "{{ .Virtenv }}/bin",
    "PATH": "{{ .Virtenv }}/bin:$PATH"
  }
```

---

I think I already tried this – you can see that I changed it [here](https://github.com/cultureamp/ca-llm-gateway/commit/8e64157eb448050fad16ee892fa2e55bdb65d54c) when pulling out the local versions of the plugins. This was presumably because something didn't work with setting in the path directly (but this is what I was doing before using plugins). I can't remember *what* didn't work though...

---

@jay-aye-see-kay ☝️  I just checked, and I can confirm that it doesn't get added to path by direct manipulation when the app's `devbox.json` (or maybe another plugin?) also manipulates `$PATH` – it seems the app one takes precedence, rather than both being run.

---

Oh gross, I hadn't seen that issue before - IMO that's a bug in devbox, I'll make some time this week to raise that with them. That's likely an issue for some of the built in plugins, i.e. the ruby plugin uses this approach https://github.com/jetify-com/devbox/blob/main/plugins/ruby.json

Let's leave it as is, it works and there's no downside the the `init_hook` approach (aside from being a little verbose).

---

Bug report here: https://github.com/jetify-com/devbox/issues/2138